### ISSUE TITLE
[3.100 remove legacy workaround for this] Add a result field to tasks

### ISSUE THREAD

**Issue by mdellweg** [pulp/pulpcore#6079]
edit: The deprecation introduced with this should turn into an error with 3.100.

The way we currently need to infer what a task that is basically a deferred api view actually did for us is weird.
We should use the serializer from the viewset that dispatched the task and serialize the result into the tasks "result" field.
This would make the "typical task" much more like a "deferred view". (scheduled tasks can still leave the result empty).

This relates to:
https://github.com/pulp/pulpcore/issues/2028
https://github.com/pulp/pulpcore/issues/3771

**Comment by quba42** [comment]
Does this mean users waiting on a task result will no longer need to parse the hrefs in `created_resources` to figure out what type of things were created?
